### ISSUE TITLE
Consider an explicit "read receipt" for messages sent to Hopskotch

### ISSUE THREAD

**Issue by bfc5288** [scimma/hop-client#141]
### Description
As a user, I would like a confirmation that a message sent with `hop publish` is received by Hopskotch, i.e. a form of "read receipt" for messages being published.

This may already be (indirectly) fulfilled by `hop-client` throwing a timeout if the client is unable to connect to Hopskotch. Otherwise, `hop-client` returns after the message receipt, which implicitly indicates that the message was sent.

Would adding an explicit indication of receipt (e.g., via Kafka on-delivery message callback) be useful/feasible? @spenczar @cnweaver Any thoughts on this?

**Comment by spenczar** [comment]
At the very lowest level, `confluent-kafka-python`'s `produce` API is asynchronous, so this would have to be done with a callback, or by calling `flush()` which blocks until all messages are sent.

`adc-streaming` supports setting a callback in a [`ProducerConfig`'s `delivery_callback`](https://github.com/astronomy-commons/adc-streaming/blob/4e489f8f944b1762a2654dfa74bfbc2f42d62df3/adc/producer.py#L71) attribute.

It might be possible to block on the callback firing, but I'm not sure exactly, since the callback isn't a Python future or coroutine or anything - it's called from a background thread from inside librdkafka.

It might be easier to expose a `flush` method on the `hop.io.Producer` class. If a caller wants to be sure a message has been sent, they can call `flush()`; it should return when the message is sent.

If an error occurs during the flush, though, then the error will only be communicated through an error callback, so they may be in the same boat.

Short version is that confluent-kafka-python will make this somewhat unsatisfying, I think, but we may be able to do something.
